src/khepri_tree.hrl

```diff
@@ -24,5 +24,4 @@
 
 -record(tree, {root = #node{} :: khepri_tree:tree_node(),
                keep_while_conds = #{} :: khepri_tree:keep_while_conds_map(),
-               keep_while_conds_revidx = #{} ::
-               khepri_tree:keep_while_conds_revidx()}).
+               keep_while_conds_revidx = #{}}).
```

Comment:
Shouldn’t the type be left unmodified?

Reply:
After testing to restore the type spec, Dialyzer complains because the type is opaque. I believe the complaint is valid. There is room for improvement in the type specs around `tree()` but it is out of scope for this pull request.

Meanwhile, can you use `-type` instead of `-opaque` for `khepri_tree:keep_while_conds_revidx()`?